### ISSUE TITLE
Renovate app disabled on gitlab.com

### ISSUE THREAD

**Issue by stephenwade** [renovatebot/renovate#7843]
# For more information please checkout
* https://docs.renovatebot.com/install-gitlab-app
* https://docs.renovatebot.com/gitlab-bot-security
* https://gitlab.com/renovate-bot/renovate-runner

---
<details>
<summary>Old description</summary>
**What Renovate type, platform and version are you using?**

GitLab hosted app

**Describe the bug**

After I pushed [this commit](https://gitlab.com/stephenwade/festival/-/commit/edb9bcce3d57d8c95be45282414e292ad45cd55b) to one of my GitLab projects, Renovate started failing, as seen in the screenshot below. The other problem is that I can't access the logs. When I click on any of those logs, nothing loads and I see a 404 error in the JavaScript console.

I tried to reproduce the problem locally, but my local copy of Renovate ran with no issues (that's why the current merge requests on that project are opened by me, not Renovate). I think I need help investigating on your server end.

<img width="760" alt="Screenshot of the Renovate app. All the logs are labeled as 'error'." src="https://user-images.githubusercontent.com/4148577/100549715-a9a00900-3242-11eb-83c2-968263092748.png">

**Relevant debug logs**

I can't access the logs. When I click on any of the log entries in the Renovate app, nothing loads and I see a 404 error in the JavaScript console.

**To Reproduce**

I don't know.

**Additional context**

I tried running it locally against the same repository, but it worked fine. I don't know what else to try.
</details>

**Comment by viceice** [comment]
Renovate app is currently disabled on gitlab. We will provide more information later.

**Comment by vnea** [comment]
@viceice 
Is Renovate App also disabled for gitlab.com for self hosted Renovate App ?

**Comment by rarkins** [comment]
No, this is just for the hosted app. If you self-host Renovate, you are not affected. We should have an update tomorrow regarding the app.

**Comment by vnea** [comment]
> No, this is just for the hosted app. If you self-host Renovate, you are not affected. We should have an update tomorrow regarding the app.

Thank you :)

**Comment by toinehartman** [comment]
@rankins Is there already any news on this?

**Comment by rarkins** [comment]
Please see https://github.com/renovatebot/renovate/blob/master/docs/usage/install-gitlab-app.md for an explanation.

**Comment by Potherca** [comment]
To save people from "yet another click":tm:, the linked document (at the time of this writing, 2020-12-03) states:

> The hosted Renovate App for GitLab is currently offline and will be restored later either as an OAuth app or once a viable security model for bots on GitLab.com is available. For more details on GitLab security for bots, please see the GitLab Bot Security doc.

**Comment by caiofbpa** [comment]
There's also very easy instructions on how to setup your own renovatebot runner inside GitLab, using GitLabCI infrastructure, minimal configuration and no security risk: https://gitlab.com/renovate-bot/renovate-runner/-/blob/master/README.md

**Comment by rarkins** [comment]
@Potherca that's quite unhelpful because you have left out the link to the main doc discussing GitLab security and a migration plan. I am going to minimize your comment as it does more harm than help.

**Comment by rarkins** [comment]
The important information is linked from the above doc to here: https://github.com/renovatebot/renovate/blob/master/docs/usage/gitlab-bot-security.md

**Comment by leaumar** [comment]
> There's also very easy instructions on how to setup your own renovatebot runner inside GitLab, using GitLabCI infrastructure, minimal configuration and no security risk: https://gitlab.com/renovate-bot/renovate-runner/-/blob/master/README.md

I tried to follow this guide, but it didn't get me there:

- created a new blank project
- added .gitlab-ci.yml containing:
```
include:
  - project: 'renovate-bot/renovate-runner'
    file: '/templates/renovate.gitlab-ci.yml'
```
- set the `RENOVATE_TOKEN` ci variable with a new token with the right permissions (left the rest default, so a shared runner, etc)

Trying to then run the CI just yields
> The form contains the following error:
> No stages / jobs for this pipeline.

I've only used basic CI setups before, so I'm assuming this yml should import the renovate.gitlab-ci.yml file from the renovate-runner project and _that_ should define the necessary stages/jobs. It seems something isn't set up right or the instructions omit something, or did I misinterpret something in the guide?

**Comment by ChristianIvicevic** [comment]
@leaumar I just looked into the pipeline files. Have you tried to manually add the following to your custom renovate CI file?

```yaml
stages:
  - deploy
```

This might fix your problem, but I cannot assure it right now. I will try to setup this myself in the next few moments, so I will try it myself as well and update the post if I know more.

**Comment by caiofbpa** [comment]
I got it to work very easily by following all steps in the README: https://gitlab.com/renovate-bot/renovate-runner/-/blob/master/README.md

Anyway, I think another issue should be created for problems in that installation, this issue is more focused on news about getting RenovateBot working again as a standalone GitLab App.

**Comment by leaumar** [comment]
> Anyway, I think another issue should be created for problems in that installation

https://gitlab.com/renovate-bot/renovate-runner/-/issues/5

**Comment by Potherca** [comment]
@rarkins I've updated my response to include the security statement/document you linked to.

Maybe I am misunderstanding things, but wouldn't adding more information _here_ rather just going RTFM -> Link be _more_ helpful?

**Comment by rarkins** [comment]
We spent the last week working with multiple involved companies to research the security risks and prepare an assessment plus migration plan. We then posted that into our docs so that they can be version controlled and kept up to date. This is not RTFM in any way and you're now 2/2 for making unhelpful and distracting comments in this thread.

If we were to copy/paste in here then we'd risk people finding out of date information in future instead of the canonical source.
